src/backends/KiboCommerce.js

```diff
@@ -0,0 +1,194 @@
+
+    return refreshedTicket;
+  }
+  async getAccessToken() {
+    // get current Kibo API auth ticket
+
+    let authTicket = await this._getAuthTicket();
+    // if no current ticket, perform auth
+    // or if ticket expired, refresh auth
+    if (!authTicket) {
+      authTicket = await this.authenticate();
+    } else if (authTicket.expires_at < Date.now()) {
+      authTicket = await this.refreshTicket(authTicket);
+    }
+
+    return authTicket.access_token;
+  }
+}
+
+export class KiboCommerce {
+  constructor({ apiHost, authHost, clientId, sharedSecret }) {
+    this.graphQLUrl = `https://${apiHost}/graphql`;
+    this.authHelper = new APIAuthenticationHelper(clientId, sharedSecret, authHost);
+  }
+  async getAccessToken() {
+    return await this.authHelper.getAccessToken();
+  }
+  async getHeaders() {
+    const authToken = await this.getAccessToken();
+    return {
+      headers: {
+        Authorization: `Bearer ${authToken}`,
+        'Content-Type': 'application/json',
+      },
+    };
+  }
+  async performSearch({ query, filter, pageSize, startIndex }) {
+    const headers = await this.getHeaders();
+    const body = {
+      query: productSearchQuery,
+      variables: { query, filter, pageSize, startIndex },
+    };
+    const params = { method: 'POST', ...headers, body: JSON.stringify(body) };
+    const request = await fetch(this.graphQLUrl, params);
+    const response = await request.json();
+    return response.data.productSearch;
+  }
+  async getItems(state, ids = []) {
+    const { PAGE_SIZE } = state;
+    const filter = ids.map((id) => `productCode eq ${id}`).join(' or ');
+    try {
+      const result = await this.performSearch({ filter, pageSize: PAGE_SIZE });
+      return this.normalizeItems(result.items);
+    } catch (error) {
+      console.error(error);
+      throw new ProductSelectorError('Could not get items', ProductSelectorError.codes.GET_SELECTED_ITEMS);
+    }
+  }
+  async search(state) {
+    const { searchText, page, PAGE_SIZE } = state;
+    try {
+      const startIndex = (page.curPage || 0) * PAGE_SIZE;
+      const result = await this.performSearch({ query: searchText, pageSize: PAGE_SIZE, startIndex });
+      if (!result.totalCount) {
+        return {
+          items: [],
+          page: { numPages: 0, curPage: 0, total: 0 },
+        };
+      }
+      return {
+        items: this.normalizeItems(result.items),
+        page: {
+          numPages: Math.ceil(result.totalCount / PAGE_SIZE),
+          curPage: page.curPage,
+          total: result.totalCount,
+        },
+      };
+    } catch (error) {
+      console.error(error);
+      throw new ProductSelectorError('Could not search', ProductSelectorError.codes.GET_ITEMS);
+    }
+  }
+  exportItem(item) {
+    return item.id;
+  }
+  getImageUrl(product) {
+    const { content = {} } = product;
+    const { productImages = [] } = content;
+    const [mainImage, ...rest] = productImages;
```

Comment:
The rest aren't used, seems a bit weird to destructure like this instead of `const mainImage = productImages[0]`.

Reply:
good catch, fixed this.